controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -388,6 +393,7 @@ func (r *StorageConsumerReconciler) reconcileCephClientRBDProvisioner() error {
 			return err
 		}
 
+		addStorageRelatedAnnotations(&r.cephClientRBDProvisioner.ObjectMeta, r.storageConsumer.Name, "rbd", "provisioner")
```

Comment:
```suggestion
		addStorageRelatedAnnotations(&r.cephClientRBDProvisioner, r.storageConsumer.Name, "rbd", "provisioner")
```

Reply:
going with this approach as discussed offline
```
if obj.Annotations == nil {
		obj.Annotations = map[string]string{}
	}
```
in this, we don't need above suggestion

Reply:
@subhamkrai I don't see the connection. Maybe you misunderstood the suggestion, `ObjectMeta` is embedded inside `cephClientRBDProvisioner` which means that when you take the address of ObjectMeta you are actually taking the address of `cephClientRBDProvisioner` which makes `&cephClientRBDProvisioner` equal to `cephClientRBDProvisioner.ObjectMeta` in this context.

What I am suggesting here is to remove the unnecessary `.ObjectMeta` to make the call more readable and the line shorter

Reply:
okay, this how it looks now 
`addStorageRelatedAnnotations(r.cephClientCephFSProvisioner, r.storageConsumer.Name, "cephfs", "provisioner")`

```
func addStorageRelatedAnnotations(cephClient *rookCephv1.CephClient, storageConsumerName, storageClaim, cephUserType string) {
	if cephClient.Annotations == nil {
		cephClient.Annotations = map[string]string{}
	}
	cephClient.Annotations[StorageConsumerAnnotation] = storageConsumerName
	cephClient.Annotations[StorageClaimAnnotation] = storageClaim
	cephClient.Annotations[StorageCephUserTypeAnnotation] = cephUserType
}
```